`Teku failed to start` due to `State must be advanced to checkpoint epoch boundary slot`

Updating our nodes, some of those ended up stuck in this state
 
Log message: `Teku failed to start`
 
Stack trace:
 ```
 java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: State must be advanced to checkpoint epoch boundary slot
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1159)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:146)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2357)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:619)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:31)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:141)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$exceptionallyCompose$34(SafeFuture.java:423)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:146)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateToAsync$20(SafeFuture.java:323)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:80)
	at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$2(AsyncRunner.java:47)
	at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:72)
	at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:124)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.lang.IllegalArgumentException: State must be advanced to checkpoint epoch boundary slot
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:143)
	at tech.pegasys.teku.spec.datastructures.state.CheckpointState.<init>(CheckpointState.java:42)
	at tech.pegasys.teku.spec.datastructures.state.CheckpointState.create(CheckpointState.java:56)
	at tech.pegasys.teku.services.beaconchain.WeakSubjectivityInitializer.validateAnchorIsWithinWeakSubjectivityPeriod(WeakSubjectivityInitializer.java:148)
	at tech.pegasys.teku.services.beaconchain.BeaconChainController.validateWeakSubjectivityPeriod(BeaconChainController.java:459)
	at tech.pegasys.teku.services.beaconchain.BeaconChainController.lambda$initialize$11(BeaconChainController.java:428)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	... 27 more
```

--- a/services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializer.java
+++ b/services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializer.java
@@ -35,6 +35,8 @@ import tech.pegasys.teku.spec.datastructures.state.CheckpointState;
 import tech.pegasys.teku.spec.datastructures.state.Fork;
 import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
 import tech.pegasys.teku.spec.datastructures.util.ChainDataLoader;
+import tech.pegasys.teku.spec.logic.common.statetransition.exceptions.EpochProcessingException;
+import tech.pegasys.teku.spec.logic.common.statetransition.exceptions.SlotProcessingException;
 import tech.pegasys.teku.storage.api.StorageQueryChannel;
 import tech.pegasys.teku.storage.api.StorageUpdateChannel;
 import tech.pegasys.teku.storage.api.WeakSubjectivityUpdate;
@@ -144,12 +146,25 @@ public class WeakSubjectivityInitializer {
       final UInt64 currentSlot,
       final Spec spec,
       final WeakSubjectivityCalculator weakSubjectivityCalculator) {
+    final UInt64 checkpointSlot =
+        spec.computeStartSlotAtEpoch(initialAnchor.getCheckpoint().getEpoch());
+    final BeaconState stateAtCheckpoint;
+    if (initialAnchor.getState().getSlot().isLessThan(checkpointSlot)) {
+      // play empty slot up to checkpoint slot
+      try {
+        stateAtCheckpoint = spec.processSlots(initialAnchor.getState(), checkpointSlot);
+      } catch (SlotProcessingException | EpochProcessingException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      stateAtCheckpoint = initialAnchor.getState();
+    }
     final CheckpointState checkpointState =
         CheckpointState.create(
             spec,
             initialAnchor.getCheckpoint(),
             initialAnchor.getBlockSummary(),
-            initialAnchor.getState());
+            stateAtCheckpoint);
     if (!weakSubjectivityCalculator.isWithinWeakSubjectivityPeriod(checkpointState, currentSlot)) {
       throw new IllegalStateException(
           "Cannot sync outside of weak subjectivity period. Consider re-syncing your node using --checkpoint-sync-url or use --ignore-weak-subjectivity-period-enabled to ignore this check.");

